source/renderer/app/components/staking/stake-pools/StakePools.js

```diff
@@ -160,50 +191,84 @@ export default class StakePools extends Component<Props, State> {
               search={search}
               onSearch={this.handleSearch}
               onClearSearch={this.handleClearSearch}
+              onGridView={this.handleGridView}
+              onListView={this.handleListView}
+              isListView={isListView}
+              isGridView={isGridView}
+              isFixed={(isFixed || maintainFixed) && !!filteredStakePoolsList.length}
               isClearTooltipOpeningDownward
+              isScrolled={isScrolled}
             />
-
-            {stakePoolsDelegatingList.length > 0 && (
+            {isListView && (
               <Fragment>
-                <h2 className={styles.listTitle}>
-                  {intl.formatMessage(messages.delegatingListTitle)}
+                <h2 className={tableHeadingClasses}>
+                  <FormattedMessage
+                    {...listTitleMessage}
+                    values={{
+                      pools: filteredStakePoolsList.length,
+                    }}
+                  />
                 </h2>
-                <StakePoolsList
-                  listName="stakePoolsDelegatingList"
-                  stakePoolsList={stakePoolsDelegatingList}
+                <StakePoolsTable
+                  listName={SELECTED_INDEX_TABLE}
+                  stakePoolsList={filteredStakePoolsList}
                   onOpenExternalLink={onOpenExternalLink}
                   currentTheme={currentTheme}
-                  isListActive={selectedList === 'stakePoolsDelegatingList'}
+                  isListActive={selectedList === SELECTED_INDEX_TABLE}
                   setListActive={this.handleSetListActive}
                   containerClassName="StakingWithNavigation_page"
                   onSelect={this.onDelegate}
                   numberOfStakePools={stakePoolsList.length}
                   showWithSelectButton
+                  onScrollView={this.handleSearchComponentScrollView}
+                  maintainFixed={maintainFixed && !!filteredStakePoolsList.length}
+                  isScrolled={isScrolled}
+                />
+              </Fragment>
+            )}
+            {isGridView && (
```

Comment:
`StakePoolsListView.js `

Reply:
I have already separated the logic on 2 components - StakePoolsTable.js & StakePoolsList.js